thirdparty/kyaml/runfn/runfn.go

```diff
@@ -192,14 +193,50 @@ func (r RunFns) runFunctions(
 		})
 	}
 
-	var err error
+	inputResources, err := input.Read()
+	if err != nil {
+		return err
+	}
+
+	selectedInput := inputResources
+
+	if !r.Selector.IsEmpty() {
+		err = fnruntime.SetResourceIds(inputResources)
+		if err != nil {
+			return err
+		}
+
+		// select the resources on which function should be applied
+		selectedInput, err = fnruntime.SelectInput(inputResources, []kptfile.Selector{r.Selector}, &fnruntime.SelectionContext{RootPackagePath: r.uniquePath})
```

Comment:
nit: each arg in newline will be easier on eyes.

Reply:
Done.